Content.Shared/Stunnable/SharedStunSystem.cs

```diff
@@ -71,6 +75,38 @@ public override void Initialize()
             SubscribeLocalEvent<StunnedComponent, PickupAttemptEvent>(OnAttempt);
             SubscribeLocalEvent<StunnedComponent, IsEquippingAttemptEvent>(OnEquipAttempt);
             SubscribeLocalEvent<StunnedComponent, IsUnequippingAttemptEvent>(OnUnequipAttempt);
+            SubscribeLocalEvent<MobStateComponent, MobStateChangedEvent>(OnMobStateChanged);
+        }
+
+
+
+        private void OnMobStateChanged(EntityUid uid, MobStateComponent component, MobStateChangedEvent args)
+        {
+            if (!TryComp<StatusEffectsComponent>(uid, out var status))
+            {
+                return;
+            }
+            switch (args.NewMobState)
+            {
+                case MobState.Alive:
```

Comment:
why is this necessary? if they died and were revived in the middle of a stun they should stay stunned imo. that or have a field on the statuseffectprototype for whether it should get removed on death/crit so this can be generalized

Reply:
That is what it is on master, and this is primarily a MobState refactor PR. I want to avoid changing anything mechanically